src/version.hpp

```diff
@@ -0,0 +1,31 @@
+#ifndef BRIDGESTAN_VERSION_HPP
+#define BRIDGESTAN_VERSION_HPP
+
+#include <string>
+
+#ifndef BRIDGESTAN_STRING_EXPAND
+#define BRIDGESTAN_STRING_EXPAND(s) #s
+#endif
+
+#ifndef BRIDGESTAN_STRING
+#define BRIDGESTAN_STRING(s) BRIDGESTAN_STRING_EXPAND(s)
```

Comment:
[question]

Why are there two functions here when the second just looks like it delegates to the first?  

Is all this complexity being introduced so we can use both `BRIDGESTAN_MAJOR` as a number and use `bridgestan::MAJOR_VERSION` as a string?   But if they're always going to be numbers, why do we need the strings?  I think the strings are useful for values like "alpha" or "beta".

Reply:
I copied what was done in [`stan/version.hpp`](https://github.com/stan-dev/stan/blob/develop/src/stan/version.hpp). The macros in that file look to be quite old, so I'm not 100% sure what the second level of indirection does for us.

Yes, the idea is that `BRIDGESTAN_MAJOR` as a preprocessor define could be used at compile time, if someone for example was writing code which did something different depending on the version of bridgestan used. The `bridgestan::MAJOR_VERSION` is the equivalent but namespaced and more runtime friendly, though I do not know why it is a string specifically.